client/lib/json-patch-lib.ts

```diff
@@ -15,74 +15,68 @@ export interface AddOperation<T> extends BaseOperation {
 
 export interface RemoveOperation extends BaseOperation {
   op: PatchOp.REMOVE
+  value?: undefined
 }
 
 export interface ReplaceOperation<T> extends BaseOperation {
   op: PatchOp.REPLACE
   value: T
 }
 
-export type Doc = { [key: string]: any }
-
 export type Operation = AddOperation<any> | RemoveOperation | ReplaceOperation<any>
 
-export function getValueByPointer (doc: any, pointer: string): any {
+export function getValueByPointer (doc: Record<string, any>, pointer: string): any {
```

Comment:
👍 Like the use of `Record`
⚒️ Could you update the name here from `doc` since we deleted that? I think `data` or `record` would be more clear. Also, update `any` to `unknown`.

Reply:
Changed to generic so reduce method works properly